src/ExtendedPdo.php

```diff
@@ -111,7 +134,11 @@ public function establishConnection(): void
         // connect
         $this->profiler->start(__FUNCTION__);
         list($dsn, $username, $password, $options, $queries) = $this->args;
-        $this->pdo = PDO::connect($dsn, $username, $password, $options);
+        if ($this->driverSpecific) {
+            $this->pdo = PDO::connect($dsn, $username, $password, $options);
```

Comment:
I thought we don't need the connect method itself as this is implemented internally in the PDO.

Reply:
I was looking at the final release of the new PDO

```PDO::__construct()``` always returns PDO and ```PDO::connect()``` returns Pdo\\{diver} or PDO  depending on available drivers

So if you use ```ExtendedPdo::getPdo()``` you would get the correct typed object.